packages/backend-core/src/sql/utils.ts

```diff
@@ -142,7 +142,7 @@ export function breakRowIdField(_id: string | { _id: string }): any[] {
 export function isInvalidISODateString(str: string) {
   const trimmedValue = str.trim()
   if (!ISO_DATE_REGEX.test(trimmedValue)) {
-    return false
+    return true
```

Comment:
This was a logic bug that could have been causing us date problems, as it is incorrectly saying invalid dates are actually valid.

Reply:
I think this happened because usually this should be `isValidISODateString`, instead of `isInvalid` 😅  I can see why this was confusing

Reply:
I wonder if it makes more sense to have this be:

```typescript
export function isInvalidISODateString(str: string) {
  return !isValidISODateString(str)
}
```

Reply:
Yea I was thinking the same. The invalid check is more thorough and actually tries constructing a date after the regex check. I'll pull the more thorough check into the `isValid`, then update isInvalid` to just invert `isValid`.